src/dashboard/src/main/management/commands/rebuild_transfer_backlog.py

```diff
@@ -0,0 +1,209 @@
+            transfer_dir = os.path.basename(directory)
+            self.stdout.write('Indexing {} ({})'.format(
+                transfer_uuid, transfer_dir))
+
+            self.process_transfer(es_client, os.path.join(transfer_backlog_dir, directory), transfer_uuid)
+            self.ss_reindex(transfer_uuid)
+
+    def process_transfer(self, es_client, transfer_path, transfer_uuid):
+        transfer_dir = os.path.basename(transfer_path)
+        tree = ElementTree.parse(os.path.join(
+            transfer_path, 'metadata/submissionDocumentation/METS.xml'))
+        root = tree.getroot()
+
+        # Attempt to insert row in MySQL Transfers table if it doesn't already
+        # exist.
+        try:
+            Transfer.objects.create(
+                uuid=transfer_uuid,
+                currentlocation=os.path.join(
+                    transfer_path.replace(django_settings.SHARED_DIRECTORY,
+                                          '%sharedPath%', 1),
+                    transfer_dir))
+        except django.db.utils.IntegrityError:
+            pass
+
+        # Extract list of files from METS data so as to populate the MySQL
+        # Files table.
+        mets_bns = '{http://www.loc.gov/METS/}'
+        files = root.findall('{mets_bns}fileSec/'
+                             '{mets_bns}fileGrp/'
+                             '{mets_bns}file'.format(mets_bns=mets_bns))
+        for file in files:
+            id_ = file.attrib['ID'][-36:]
+
+            # Extract original path.
+            location = file.find('{http://www.loc.gov/METS/}FLocat')
+            file_path = location.attrib['{http://www.w3.org/1999/xlink}href']
+
+            # Deleted archives (e.g., .zip) files will be listed in the
+            # transfer METS.xml but they will not exist on disk.
+            file_path = os.path.join(transfer_path, file_path)
+            if not os.path.exists(file_path):
+                self.warning('File not found: {}'.format(file_path))
+                continue
+
+            path_to_object = os.path.dirname(file_path)
+            file_basename = os.path.basename(file_path)
+            logger.debug("File %s, path_to_object=%s, file_basename=%s",
+                         id_, path_to_object, file_basename)
+
+            # Calculate filesize
+            real_file_path = os.path.join(
+                transfer_path, path_to_object, file_basename)
+            file_size = str(os.path.getsize(real_file_path))
+
+            location_with_token = '%transferDirectory%{}/{}'.format(
+                path_to_object,
+                file_basename)
+            checksum = self.get_file_checksum(real_file_path)
+
+            # Attempt to insert row in MySQL Files table if it doesn't already
+            # exist.
+            try:
+                File.objects.create(
+                    uuid=id_,
+                    originallocation=location_with_token,
+                    currentlocation=location_with_token,
```

Comment:
Given @jhsimpson's [comment](https://github.com/artefactual/archivematica/pull/894#issuecomment-359978178), I'm wondering if what you were doing here previously was getting the original (unsanitized) file name/path from the METS file. In any case, shouldn't we be doing that? I mean, given how this works right now, if I backlog a transfer containing files with sanitizable names, then delete File rows and rebuild the index using this command, then won't the `originallocation` value be technically incorrect. Probably a minor issue, but just pointing it out